src/shared/sleep-config.c

```diff
@@ -229,6 +245,51 @@ int find_hibernate_location(char **device, char **type, uint64_t *size, uint64_t
                                 continue;
                         }
 
+                        /* Check if swap matches the resume= and resume_offset= values if configured. */
```

Comment:
Please let this be a separate function. It'll be much clearer and error handling will be easier.